scene/resources/material.cpp

```diff
@@ -1104,14 +1104,17 @@ void BaseMaterial3D::_update_shader() {
 	}
 
 	if (distance_fade != DISTANCE_FADE_DISABLED) {
+		// For all distance fade modes, use circular fade (disatnce to the
+		// object instead of Z distance), so that the fade is always the same
+		// regardless of the camera angle.
+		// This has a small performance cost, but the quality increase is worth it.
 		if ((distance_fade == DISTANCE_FADE_OBJECT_DITHER || distance_fade == DISTANCE_FADE_PIXEL_DITHER)) {
 			if (!RenderingServer::get_singleton()->is_low_end()) {
 				code += "\t{\n";
 				if (distance_fade == DISTANCE_FADE_OBJECT_DITHER) {
-					code += "\t\tfloat fade_distance = abs((INV_CAMERA_MATRIX * WORLD_MATRIX[3]).z);\n";
-
+					code += "\t\tfloat fade_distance = sqrt(pow((INV_CAMERA_MATRIX * WORLD_MATRIX[3]).z, 2) + pow((INV_CAMERA_MATRIX * WORLD_MATRIX[3]).x, 2) + pow((INV_CAMERA_MATRIX * WORLD_MATRIX[3]).y, 2));\n";
```

Comment:
The `sqrt()` is needed to make the effective fade distance about the same as before.

Reply:
Ah, this change seems wrong. The ``.z`` should not be here. Including the ``.z`` makes it the exact same as the original (functionally speaking, it actually produces worse instructions than the original). Removing the ``.z`` makes it the same as your original edit.

<details>
<summary><b>shader ISA disassembly</b></summary>

Disassembly produced in Radeon GPU Analyzer for ``abs(float)``:

```
  v_cvt_pkrtz_f16_f32  v0, abs(s0), lit(0x3ecccccd)     // 000000000028: D52F0100 0001FE00 3ECCCCCD
```

Disassembly produced for ``length(float)``:
```
  v_mul_legacy_f32  v0, s0, s0                          // 000000000028: D5070000 00000000 // this is multiply a variable by itself
  v_sqrt_f32    v0, v0                                  // 000000000030: 7E006700 // then take the sqrt (for a float this is functionally the same as abs() but takes 2 more instructions
  v_cvt_pkrtz_f16_f32  v0, v0, lit(0x3ecccccd)          // 000000000034: D52F0000 0001FF00 3ECCCCCD
```

as you can see abs is just an input modifier so it gets packed into the 32-16 bit conversion function and is basically free, whereas length emits a multiply and sqrt instruction.
</details>